bokehjs/src/coffee/renderer/tile/tile_renderer.coffee

```diff
@@ -24,6 +29,35 @@ class TileRendererView extends PlotWidget
     @_last_height = undefined
     @_last_width = undefined
 
+  _add_attribution: () =>
+    attribution = @mget('tile_source').get('attribution')
+
+    if _.isString(attribution) and attribution.length > 0
+      if @attributionEl?
+        @attributionEl.html(attribution)
+      else
+        border_width = 1
+
+        bottom_offset = @map_plot.get('min_border_bottom') + border_width
+        right_offset = @map_plot.get('min_border_right') + border_width
+
+        @attributionEl = $('<div>')
+          .html(attribution)
+          .addClass('bk-tile-attribution')
+          .css({
+            'position': 'absolute'
+            'bottom': "#{bottom_offset}px"
+            'right': "#{right_offset}px"
+            'background-color': 'rgba(255,255,255,0.8)'
+            'font-size': '9pt'
+            'font-family': 'sans-serif'
+            'padding-left': '2px'
+            'padding-right': '2px'
```

Comment:
This could take advantage of `TextProps`. Of course this would require more work, because there would have to be `Attribution` annotation (or a generic one; I tried to use `Span`, but it wasn't helpful), then `TileSource.annotation` would have `String | Instance(Annotation)` signature.